vita3k/modules/SceShellSvc/SceShellSvc.cpp

```diff
@@ -17,6 +17,39 @@
 
 #include <module/module.h>
 
+#include <modules/module_parent.h>
+
 EXPORT(int, sceShellSvcGetSvcObj) {
-    return UNIMPLEMENTED();
+    static Ptr<void> client_vtable = Ptr<void>();
+    static Ptr<int> svc_client;
+    if (!client_vtable) {
+        // initialize the vtable
+        client_vtable = create_vtable({
+                                          0x101C93F8, // destroy
+                                          0xA22C3E01, // connect
+                                          0xEC73331C, // disconnect
+                                          0xD484D36D, // terminateConnection
+                                          0x28BD5F19, // invokeSyncMethod
+                                          0x73C72FBB, // invokeSyncMethod
+                                          0xAFD10F3B, // invokeAsyncMethod
+                                          0x387AFA3F, // invokeAsyncMethod
+                                          0xF8C2B8BA, // tryGetResult
+                                          0x4EBB01A2, // tryGetResult
+                                          0x8FF23C3C, // pollEventFlag
+                                          0x45C32034, // waitEventFlag
+                                          0x004F48ED, // getUserData
+                                          0xA3E650B0, // getMsg
+                                          0x60EFADE7, // tryGetMsg
+                                          0xA5AA193C, // ~Client
+                                      },
+            emuenv.mem);
```

Comment:
It would be preferable not to duplicate this code with `IPMI::Client::create(IPMI::Client**, IPMI::Client::Config const*, void*, void*)`

Reply:
Code is little diferent. Check return

Reply:
Yes, but the vtable is the same, it's not impossible for a program to compare vtable addresses

Reply:
I know, but return is diferent so 🤔

Reply:
Even if the return value is different, they both should use the same vtable address.

Reply:
hmm vtable is same yes, but just i no know how doring that ? 🤔 call ipmi func ?